tests/test_wandb.py

```diff
@@ -150,6 +150,55 @@ def test_custom_dir(wandb_init_run):
     assert len(glob.glob("/tmp/wandb/run-*")) > 0
 
 
+def test_login_key(local_netrc, capsys):
+    wandb.login(key="A"* 40)
+    out, err = capsys.readouterr()
+    assert "wandb: WARNING If" in err
+    assert wandb.api.api_key == "A" * 40
+
+
+def test_login_existing_key(local_netrc):
+    os.environ["WANDB_API_KEY"] = "B" * 40
+    wandb.ensure_configured()
+    wandb.login()
+    assert wandb.api.api_key == "B" * 40
+
+
+def test_login_no_key(local_netrc, mocker):
+    if os.environ.get("WANDB_API_KEY"):
```

Comment:
Maybe this should be a fixture that we autouse to ensure that the API key is always cleared before these tests?